tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers.ts

```diff
@@ -674,6 +675,102 @@ const reducer = createReducer(
       };
     }
   ),
+  ////////////////////////////////////////////////
+  // Reducers related to graph structures.      //
+  ////////////////////////////////////////////////
+  on(
+    actions.graphOpFocused,
+    (
+      state: DebuggerState,
+      data: {graph_id: string; op_name: string}
+    ): DebuggerState => {
+      return {
+        ...state,
+        graphs: {
+          ...state.graphs,
+          focusedOp: {
+            graphId: data.graph_id,
+            opName: data.op_name,
+          },
+        },
+      };
+    }
+  ),
+  on(
+    actions.graphOpInfoRequested,
+    (
+      state: DebuggerState,
+      data: {graph_id: string; op_name: string}
+    ): DebuggerState => {
+      const {graph_id, op_name} = data;
+      const newState: DebuggerState = {
+        ...state,
+        graphs: {
+          ...state.graphs,
+          loadingOps: {
+            ...state.graphs.loadingOps,
+          },
+        },
+      };
+      if (newState.graphs.loadingOps[graph_id] === undefined) {
+        newState.graphs.loadingOps[graph_id] = {};
+      }
+      if (newState.graphs.loadingOps[graph_id][op_name] === undefined) {
+        newState.graphs.loadingOps[graph_id][op_name] = DataLoadState.LOADING;
+      }
+      return newState;
+    }
+  ),
+  on(
+    actions.graphOpInfoLoaded,
+    (state: DebuggerState, data): DebuggerState => {
+      const opInfo: GraphOpInfo = data.graphOpInfoResponse;
+      const {inputs, consumers, graph_ids} = opInfo;
+      const graphId = graph_ids[graph_ids.length - 1];
+      const newState: DebuggerState = {
+        ...state,
+        graphs: {
+          ...state.graphs,
+          ops: {
+            ...state.graphs.ops,
+          },
+          loadingOps: {
+            ...state.graphs.loadingOps,
+          },
+        },
+      };
+      for (const input of inputs) {
+        if (!input.data) {
+          continue;
+        }
+        if (newState.graphs.ops[graphId] === undefined) {
+          newState.graphs.ops[graphId] = {};
+        }
+        newState.graphs.ops[graphId][input.op_name] = input.data;
+        delete input.data;
```

Comment:
why do you need to modify `input`?

Reply:
Added a comment to explain that. The `input` here is a field of `opInfo`. `opInfo` is put into `newState.graphs.ops[graphId][opInfo.op_name]` below. We don't want the store
to contain duplicate data between the op itself (described by `opInfo`) and the input op
(described by `input`).

Reply:
I guess I am still lost as to why we are modifying the action payload (this delete call is only modifying the `inputs` which is from the `data` (really, action payload), right?) To ease my confusion, can you point me to a test that covers what you just said?

Reply:
I agree that mutating the data payload is suboptimal, even though it doesn't cause any us any harm right away.

I've made the following revision. 
1. In stead of calling `delete` on the payload, I now shallow copy the payload into another object before using it.
    The shallow copy skips the things we don't want. So there is no more need for `delete`. (new L766-L782).
2. I added a few lines in the unit test to reflect the expected behavior.